var xe=Object.defineProperty;var Oe=(r,t,e)=>t in r?xe(r,t,{enumerable:!0,configurable:!0,writable:!0,value:e}):r[t]=e;var Pe=(r,t,e)=>Oe(r,typeof t!="symbol"?t+"":t,e);import{a as ce,b as Ae}from"./paths.3EtBHCFv.js";import{s as Y,e as fe,u as ue,g as pe,f as he,n as le,a as Ie}from"./scheduler.BfJEPAwo.js";import{S as X,i as Z,e as j,c as U,a as N,d as c,k as q,g as p,o as y,p as v,s as S,f as x,h as D,l as k,m as P,n as A,q as I,t as g,b as w,A as Se,j as me}from"./index.BWJqtTvX.js";function Wt(r){return(r==null?void 0:r.length)!==void 0?r:Array.from(r)}function Le(r){let t,e;const n=r[1].default,i=fe(n,r,r[0],null);return{c(){t=j("p"),i&&i.c(),this.h()},l(s){t=U(s,"P",{class:!0});var o=N(t);i&&i.l(o),o.forEach(c),this.h()},h(){q(t,"class","font-sans text-base text-justify py-2 text-purple-950")},m(s,o){p(s,t,o),i&&i.m(t,null),e=!0},p(s,[o]){i&&i.p&&(!e||o&1)&&ue(i,n,s,s[0],e?he(n,s[0],o,null):pe(s[0]),null)},i(s){e||(y(i,s),e=!0)},o(s){v(i,s),e=!1},d(s){s&&c(t),i&&i.d(s)}}}function ze(r,t,e){let{$$slots:n={},$$scope:i}=t;return r.$$set=s=>{"$$scope"in s&&e(0,i=s.$$scope)},[i,n]}class C extends X{constructor(t){super(),Z(this,t,ze,Le,Y,{})}}class $e{constructor(t,e,n,i,s,o){Pe(this,"identifier");this.title=t,this.subtitle=e,this.date=n,this.heroImage=i,this.intro=s,this.body=o,this.identifier=this.title.replaceAll(" ","_")}}function Ee(r){let t,e;const n=r[3].default,i=fe(n,r,r[2],null);return{c(){t=j("a"),i&&i.c(),this.h()},l(s){t=U(s,"A",{class:!0,href:!0,target:!0});var o=N(t);i&&i.l(o),o.forEach(c),this.h()},h(){q(t,"class","font-sans text-base text-justify py-2 text-purple-900/70 hover:text-purple-900 underline decoration-amber-400"),q(t,"href",r[0]),q(t,"target",r[1])},m(s,o){p(s,t,o),i&&i.m(t,null),e=!0},p(s,[o]){i&&i.p&&(!e||o&4)&&ue(i,n,s,s[2],e?he(n,s[2],o,null):pe(s[2]),null),(!e||o&1)&&q(t,"href",s[0]),(!e||o&2)&&q(t,"target",s[1])},i(s){e||(y(i,s),e=!0)},o(s){v(i,s),e=!1},d(s){s&&c(t),i&&i.d(s)}}}function De(r,t,e){let{$$slots:n={},$$scope:i}=t,{href:s}=t,{target:o="_self"}=t;return r.$$set=f=>{"href"in f&&e(0,s=f.href),"target"in f&&e(1,o=f.target),"$$scope"in f&&e(2,i=f.$$scope)},[s,o,i,n]}class Q extends X{constructor(t){super(),Z(this,t,De,Ee,Y,{href:0,target:1})}}function Be(r){let t,e,n,i;const s=r[1].default,o=fe(s,r,r[0],null);return{c(){t=j("h4"),o&&o.c(),e=S(),n=j("div"),this.h()},l(f){t=U(f,"H4",{class:!0});var $=N(t);o&&o.l($),e=x($),n=U($,"DIV",{class:!0}),N(n).forEach(c),$.forEach(c),this.h()},h(){q(n,"class","h-px bg-amber-400 w-1/2 mt-1"),q(t,"class","font-sans text-xl text-left py-2 text-purple-900 my-4")},m(f,$){p(f,t,$),o&&o.m(t,null),D(t,e),D(t,n),i=!0},p(f,[$]){o&&o.p&&(!i||$&1)&&ue(o,s,f,f[0],i?he(s,f[0],$,null):pe(f[0]),null)},i(f){i||(y(o,f),i=!0)},o(f){v(o,f),i=!1},d(f){f&&c(t),o&&o.d(f)}}}function qe(r,t,e){let{$$slots:n={},$$scope:i}=t;return r.$$set=s=>{"$$scope"in s&&e(0,i=s.$$scope)},[i,n]}class K extends X{constructor(t){super(),Z(this,t,qe,Be,Y,{})}}function Ce(r){let t;return{c(){t=g(`This was a fantastic chance to share the progress and initial findings from my work and to connect\r
	with other bioinformatics researchers working on related challenges.`)},l(e){t=w(e,`This was a fantastic chance to share the progress and initial findings from my work and to connect\r
	with other bioinformatics researchers working on related challenges.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Me(r){let t;return{c(){t=g("The Problem: Reducing Errors in Barcode Synthesis")},l(e){t=w(e,"The Problem: Reducing Errors in Barcode Synthesis")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function je(r){let t;return{c(){t=g(`Spatial transcriptomics is a powerful technique for mapping gene expression across tissue samples.\r
	It relies on applying transcripts to an array with 1024×768 features, where each transcript is\r
	tagged with a unique DNA barcode. After sequencing, these barcodes are used to trace the spatial\r
	origin of each transcript.`)},l(e){t=w(e,`Spatial transcriptomics is a powerful technique for mapping gene expression across tissue samples.\r
	It relies on applying transcripts to an array with 1024×768 features, where each transcript is\r
	tagged with a unique DNA barcode. After sequencing, these barcodes are used to trace the spatial\r
	origin of each transcript.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ue(r){let t;return{c(){t=g(`However, the process of synthesizing barcodes on such arrays—using photolithography—can introduce\r
	significant errors. It’s estimated that up to 20% of bases may contain synthesis errors, often due\r
	to neighbor effects caused by scattered or diffracted light during the synthesis process. These\r
	errors make it difficult to correctly assign transcripts to their corresponding barcodes,\r
	complicating the gene expression analysis.`)},l(e){t=w(e,`However, the process of synthesizing barcodes on such arrays—using photolithography—can introduce\r
	significant errors. It’s estimated that up to 20% of bases may contain synthesis errors, often due\r
	to neighbor effects caused by scattered or diffracted light during the synthesis process. These\r
	errors make it difficult to correctly assign transcripts to their corresponding barcodes,\r
	complicating the gene expression analysis.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Re(r){let t;return{c(){t=g("Our Approach: Optimizing Barcode Layouts")},l(e){t=w(e,"Our Approach: Optimizing Barcode Layouts")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function He(r){let t;return{c(){t=g(`In my talk, I discussed how our research addresses this problem by focusing on barcode layout\r
	optimization. The goal is to minimize the similarity between neighboring barcodes, reducing the\r
	likelihood of synthesis errors affecting adjacent features on the array. This gives rise to a\r
	complex combinatorial optimization problem.`)},l(e){t=w(e,`In my talk, I discussed how our research addresses this problem by focusing on barcode layout\r
	optimization. The goal is to minimize the similarity between neighboring barcodes, reducing the\r
	likelihood of synthesis errors affecting adjacent features on the array. This gives rise to a\r
	complex combinatorial optimization problem.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ge(r){let t;return{c(){t=g(`We demonstrated that this problem is MaxSNP-hard, meaning it’s computationally challenging to\r
	solve optimally or even to approximate. While Integer Linear Programming (ILP) formulations could,\r
	in theory, yield optimal solutions, they are only feasible for very small instances. For larger\r
	arrays, like the ones used in spatial transcriptomics, we turned to heuristic algorithms capable\r
	of computing near-optimal layouts.`)},l(e){t=w(e,`We demonstrated that this problem is MaxSNP-hard, meaning it’s computationally challenging to\r
	solve optimally or even to approximate. While Integer Linear Programming (ILP) formulations could,\r
	in theory, yield optimal solutions, they are only feasible for very small instances. For larger\r
	arrays, like the ones used in spatial transcriptomics, we turned to heuristic algorithms capable\r
	of computing near-optimal layouts.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ne(r){let t;return{c(){t=g("Heuristic Solutions and GPU Acceleration")},l(e){t=w(e,"Heuristic Solutions and GPU Acceleration")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function We(r){let t,e,n="<li>A sorting-based algorithm</li> <li>A greedy algorithm</li> <li>A genetic algorithm</li>",i;return{c(){t=g(`Our research has developed several heuristic algorithms, including:\r
	`),e=j("ul"),e.innerHTML=n,i=g(`\r
	The most promising results so far come from the greedy algorithm, which we accelerated using GPU processing.\r
	This GPU-accelerated approach significantly reduced runtime while still producing high-quality layouts.\r
	For large-scale arrays with 1024×768 features, the greedy algorithm achieved a 35-40% improvement in\r
	layout quality over random layouts, with solutions computed in about 15 seconds. These results suggest\r
	that our methods could be applicable to real-world scenarios where speed and accuracy are critical.`),this.h()},l(s){t=w(s,`Our research has developed several heuristic algorithms, including:\r
	`),e=U(s,"UL",{class:!0,"data-svelte-h":!0}),Se(e)!=="svelte-44mn7l"&&(e.innerHTML=n),i=w(s,`\r
	The most promising results so far come from the greedy algorithm, which we accelerated using GPU processing.\r
	This GPU-accelerated approach significantly reduced runtime while still producing high-quality layouts.\r
	For large-scale arrays with 1024×768 features, the greedy algorithm achieved a 35-40% improvement in\r
	layout quality over random layouts, with solutions computed in about 15 seconds. These results suggest\r
	that our methods could be applicable to real-world scenarios where speed and accuracy are critical.`),this.h()},h(){q(e,"class","list-disc pl-8")},m(s,o){p(s,t,o),p(s,e,o),p(s,i,o)},p:le,d(s){s&&(c(t),c(e),c(i))}}}function Fe(r){let t;return{c(){t=g("Reflections on the Presentation")},l(e){t=w(e,"Reflections on the Presentation")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ve(r){let t;return{c(){t=g(`The response to my talk was very encouraging. Attendees expressed interest in the potential\r
	applications of this research to other areas of bioinformatics and gene expression analysis. There\r
	were also great questions and discussions about possible future directions, including how further\r
	refinements in the heuristic algorithms might push the performance even higher.`)},l(e){t=w(e,`The response to my talk was very encouraging. Attendees expressed interest in the potential\r
	applications of this research to other areas of bioinformatics and gene expression analysis. There\r
	were also great questions and discussions about possible future directions, including how further\r
	refinements in the heuristic algorithms might push the performance even higher.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Qe(r){let t;return{c(){t=g(`Presenting at the 5th Central German Meeting on Bioinformatics was a valuable experience, allowing\r
	me to gain feedback from the broader research community and further develop ideas for the next\r
	phase of the project. I'm looking forward to exploring additional avenues for improving barcode\r
	layout optimization and seeing how these methods might impact the field of spatial\r
	transcriptomics.`)},l(e){t=w(e,`Presenting at the 5th Central German Meeting on Bioinformatics was a valuable experience, allowing\r
	me to gain feedback from the broader research community and further develop ideas for the next\r
	phase of the project. I'm looking forward to exploring additional avenues for improving barcode\r
	layout optimization and seeing how these methods might impact the field of spatial\r
	transcriptomics.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Je(r){let t;return{c(){t=g("Acknowledgements")},l(e){t=w(e,"Acknowledgements")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ke(r){let t;return{c(){t=g(`I’d like to express my gratitude to my supervisors, Prof. Dr. Matthias Müller-Hannemann and Dr.\r
	Steffen Schüler and Antonia Schmidt for the great collaboration. This work would not have been\r
	possible without their expertise and encouragement.`)},l(e){t=w(e,`I’d like to express my gratitude to my supervisors, Prof. Dr. Matthias Müller-Hannemann and Dr.\r
	Steffen Schüler and Antonia Schmidt for the great collaboration. This work would not have been\r
	possible without their expertise and encouragement.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ye(r){let t;return{c(){t=g("The website of the 5th Central German Meeting on Bioinformatics.")},l(e){t=w(e,"The website of the 5th Central German Meeting on Bioinformatics.")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Xe(r){let t;return{c(){t=g("The slides of the talk.")},l(e){t=w(e,"The slides of the talk.")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ze(r){let t,e,n,i,s,o,f,$;return i=new Q({props:{href:"https://mittelerde-meeting.uni-jena.de/",target:"_blank",$$slots:{default:[Ye]},$$scope:{ctx:r}}}),f=new Q({props:{href:`${ce}/media/mittelerde24/presentation.pdf`,target:"_blank",$$slots:{default:[Xe]},$$scope:{ctx:r}}}),{c(){t=g(`Additional Information:\r
	`),e=j("ul"),n=j("li"),k(i.$$.fragment),s=S(),o=j("li"),k(f.$$.fragment),this.h()},l(h){t=w(h,`Additional Information:\r
	`),e=U(h,"UL",{class:!0});var b=N(e);n=U(b,"LI",{});var u=N(n);P(i.$$.fragment,u),u.forEach(c),s=x(b),o=U(b,"LI",{});var _=N(o);P(f.$$.fragment,_),_.forEach(c),b.forEach(c),this.h()},h(){q(e,"class","list-disc pl-8")},m(h,b){p(h,t,b),p(h,e,b),D(e,n),A(i,n,null),D(e,s),D(e,o),A(f,o,null),$=!0},p(h,b){const u={};b&1&&(u.$$scope={dirty:b,ctx:h}),i.$set(u);const _={};b&1&&(_.$$scope={dirty:b,ctx:h}),f.$set(_)},i(h){$||(y(i.$$.fragment,h),y(f.$$.fragment,h),$=!0)},o(h){v(i.$$.fragment,h),v(f.$$.fragment,h),$=!1},d(h){h&&(c(t),c(e)),I(i),I(f)}}}function et(r){let t,e,n,i,s,o,f,$,h,b,u,_,T,O,L,B,z,M,E,G,R,W,H,J,l,d,F,ee,V,te;return t=new C({props:{$$slots:{default:[Ce]},$$scope:{ctx:r}}}),n=new K({props:{$$slots:{default:[Me]},$$scope:{ctx:r}}}),s=new C({props:{$$slots:{default:[je]},$$scope:{ctx:r}}}),f=new C({props:{$$slots:{default:[Ue]},$$scope:{ctx:r}}}),h=new K({props:{$$slots:{default:[Re]},$$scope:{ctx:r}}}),u=new C({props:{$$slots:{default:[He]},$$scope:{ctx:r}}}),T=new C({props:{$$slots:{default:[Ge]},$$scope:{ctx:r}}}),L=new K({props:{$$slots:{default:[Ne]},$$scope:{ctx:r}}}),z=new C({props:{$$slots:{default:[We]},$$scope:{ctx:r}}}),E=new K({props:{$$slots:{default:[Fe]},$$scope:{ctx:r}}}),R=new C({props:{$$slots:{default:[Ve]},$$scope:{ctx:r}}}),H=new C({props:{$$slots:{default:[Qe]},$$scope:{ctx:r}}}),l=new K({props:{$$slots:{default:[Je]},$$scope:{ctx:r}}}),F=new C({props:{$$slots:{default:[Ke]},$$scope:{ctx:r}}}),V=new C({props:{$$slots:{default:[Ze]},$$scope:{ctx:r}}}),{c(){k(t.$$.fragment),e=S(),k(n.$$.fragment),i=S(),k(s.$$.fragment),o=S(),k(f.$$.fragment),$=S(),k(h.$$.fragment),b=S(),k(u.$$.fragment),_=S(),k(T.$$.fragment),O=S(),k(L.$$.fragment),B=S(),k(z.$$.fragment),M=S(),k(E.$$.fragment),G=S(),k(R.$$.fragment),W=S(),k(H.$$.fragment),J=S(),k(l.$$.fragment),d=S(),k(F.$$.fragment),ee=S(),k(V.$$.fragment)},l(a){P(t.$$.fragment,a),e=x(a),P(n.$$.fragment,a),i=x(a),P(s.$$.fragment,a),o=x(a),P(f.$$.fragment,a),$=x(a),P(h.$$.fragment,a),b=x(a),P(u.$$.fragment,a),_=x(a),P(T.$$.fragment,a),O=x(a),P(L.$$.fragment,a),B=x(a),P(z.$$.fragment,a),M=x(a),P(E.$$.fragment,a),G=x(a),P(R.$$.fragment,a),W=x(a),P(H.$$.fragment,a),J=x(a),P(l.$$.fragment,a),d=x(a),P(F.$$.fragment,a),ee=x(a),P(V.$$.fragment,a)},m(a,m){A(t,a,m),p(a,e,m),A(n,a,m),p(a,i,m),A(s,a,m),p(a,o,m),A(f,a,m),p(a,$,m),A(h,a,m),p(a,b,m),A(u,a,m),p(a,_,m),A(T,a,m),p(a,O,m),A(L,a,m),p(a,B,m),A(z,a,m),p(a,M,m),A(E,a,m),p(a,G,m),A(R,a,m),p(a,W,m),A(H,a,m),p(a,J,m),A(l,a,m),p(a,d,m),A(F,a,m),p(a,ee,m),A(V,a,m),te=!0},p(a,[m]){const ne={};m&1&&(ne.$$scope={dirty:m,ctx:a}),t.$set(ne);const se={};m&1&&(se.$$scope={dirty:m,ctx:a}),n.$set(se);const re={};m&1&&(re.$$scope={dirty:m,ctx:a}),s.$set(re);const ae={};m&1&&(ae.$$scope={dirty:m,ctx:a}),f.$set(ae);const ie={};m&1&&(ie.$$scope={dirty:m,ctx:a}),h.$set(ie);const oe={};m&1&&(oe.$$scope={dirty:m,ctx:a}),u.$set(oe);const de={};m&1&&(de.$$scope={dirty:m,ctx:a}),T.$set(de);const ge={};m&1&&(ge.$$scope={dirty:m,ctx:a}),L.$set(ge);const we={};m&1&&(we.$$scope={dirty:m,ctx:a}),z.$set(we);const be={};m&1&&(be.$$scope={dirty:m,ctx:a}),E.$set(be);const _e={};m&1&&(_e.$$scope={dirty:m,ctx:a}),R.$set(_e);const ye={};m&1&&(ye.$$scope={dirty:m,ctx:a}),H.$set(ye);const ve={};m&1&&(ve.$$scope={dirty:m,ctx:a}),l.$set(ve);const Te={};m&1&&(Te.$$scope={dirty:m,ctx:a}),F.$set(Te);const ke={};m&1&&(ke.$$scope={dirty:m,ctx:a}),V.$set(ke)},i(a){te||(y(t.$$.fragment,a),y(n.$$.fragment,a),y(s.$$.fragment,a),y(f.$$.fragment,a),y(h.$$.fragment,a),y(u.$$.fragment,a),y(T.$$.fragment,a),y(L.$$.fragment,a),y(z.$$.fragment,a),y(E.$$.fragment,a),y(R.$$.fragment,a),y(H.$$.fragment,a),y(l.$$.fragment,a),y(F.$$.fragment,a),y(V.$$.fragment,a),te=!0)},o(a){v(t.$$.fragment,a),v(n.$$.fragment,a),v(s.$$.fragment,a),v(f.$$.fragment,a),v(h.$$.fragment,a),v(u.$$.fragment,a),v(T.$$.fragment,a),v(L.$$.fragment,a),v(z.$$.fragment,a),v(E.$$.fragment,a),v(R.$$.fragment,a),v(H.$$.fragment,a),v(l.$$.fragment,a),v(F.$$.fragment,a),v(V.$$.fragment,a),te=!1},d(a){a&&(c(e),c(i),c(o),c($),c(b),c(_),c(O),c(B),c(M),c(G),c(W),c(J),c(d),c(ee)),I(t,a),I(n,a),I(s,a),I(f,a),I(h,a),I(u,a),I(T,a),I(L,a),I(z,a),I(E,a),I(R,a),I(H,a),I(l,a),I(F,a),I(V,a)}}}let tt=class extends X{constructor(t){super(),Z(this,t,null,et,Y,{})}};const nt=new $e("Talk at Jena University","Barcode Layout Optimization in Spatial Transcriptomics",new Date(Date.parse("1 March 2024")),`${ce}/media/mittelerde24/speaking_landscape.jpg`,`I recently had the opportunity to present my research at the 5th Central German Meeting on Bioinformatics,
	hosted by the University of Jena. My talk, titled "Barcode Layout Optimization in Spatial Transcriptomics:
	Theory and Experiments," focused on a novel approach to optimizing barcode layouts to reduce errors in spatial transcriptomics.
	This was a fantastic chance to share the progress and initial findings from my work and to connect
	with other bioinformatics researchers working on related challenges.`,tt);function st(r){let t;return{c(){t=g("Anke Tornow")},l(e){t=w(e,"Anke Tornow")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function rt(r){let t,e,n,i;return e=new Q({props:{href:"https://anketornow.de/",target:"_blank",$$slots:{default:[st]},$$scope:{ctx:r}}}),{c(){t=g(`This project, developed for the Kiel University in collaboration with multimedia designer\r
	`),k(e.$$.fragment),n=g(`\r
	, aims to showcase the intricate connections between human societies, environments, and\r
	cultures over thousands of years. Our team has worked hard to create a portal that both experts\r
	and the general public can explore with ease.`)},l(s){t=w(s,`This project, developed for the Kiel University in collaboration with multimedia designer\r
	`),P(e.$$.fragment,s),n=w(s,`\r
	, aims to showcase the intricate connections between human societies, environments, and\r
	cultures over thousands of years. Our team has worked hard to create a portal that both experts\r
	and the general public can explore with ease.`)},m(s,o){p(s,t,o),A(e,s,o),p(s,n,o),i=!0},p(s,o){const f={};o&1&&(f.$$scope={dirty:o,ctx:s}),e.$set(f)},i(s){i||(y(e.$$.fragment,s),i=!0)},o(s){v(e.$$.fragment,s),i=!1},d(s){s&&(c(t),c(n)),I(e,s)}}}function at(r){let t;return{c(){t=g(`ROOTS Compass provides a unique entry point into the research, allowing users to navigate a\r
	virtual globe to explore projects spread across continents, from ancient Pompeii to medieval North\r
	Rhine-Westphalia. By integrating various filters—such as historical periods, scientific\r
	disciplines, and research methods—users can uncover fascinating interdisciplinary links across\r
	different projects. Whether you're interested in archaeological excavations, DNA analysis, or\r
	computer modeling, the portal reveals the rich diversity of approaches used in the research at the\r
	ROOTS cluster.`)},l(e){t=w(e,`ROOTS Compass provides a unique entry point into the research, allowing users to navigate a\r
	virtual globe to explore projects spread across continents, from ancient Pompeii to medieval North\r
	Rhine-Westphalia. By integrating various filters—such as historical periods, scientific\r
	disciplines, and research methods—users can uncover fascinating interdisciplinary links across\r
	different projects. Whether you're interested in archaeological excavations, DNA analysis, or\r
	computer modeling, the portal reveals the rich diversity of approaches used in the research at the\r
	ROOTS cluster.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function it(r){let t;return{c(){t=g(`For me, as the software developer for ROOTS Compass, this project has been a rewarding challenge.\r
	Creating an interactive frontend that integrates geo data visualization with highly connected data\r
	structures was both exciting and complex. Balancing user-friendly design with the dynamic\r
	exploration of connected information pushed us to innovate, especially in handling intricate\r
	project relationships and real-time map performance.`)},l(e){t=w(e,`For me, as the software developer for ROOTS Compass, this project has been a rewarding challenge.\r
	Creating an interactive frontend that integrates geo data visualization with highly connected data\r
	structures was both exciting and complex. Balancing user-friendly design with the dynamic\r
	exploration of connected information pushed us to innovate, especially in handling intricate\r
	project relationships and real-time map performance.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function ot(r){let t;return{c(){t=g("www.roots-compass.org")},l(e){t=w(e,"www.roots-compass.org")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function lt(r){let t,e,n,i;return e=new Q({props:{target:"_blank",href:"https://www.roots-compass.org",$$slots:{default:[ot]},$$scope:{ctx:r}}}),{c(){t=g(`I'm proud to be part of this project, which not only makes cutting-edge research more accessible\r
	but also invites users to explore and engage with the complexities of our shared past. You can\r
	experience it yourself by visiting the portal at\r
\r
	`),k(e.$$.fragment),n=g(`\r
	.`)},l(s){t=w(s,`I'm proud to be part of this project, which not only makes cutting-edge research more accessible\r
	but also invites users to explore and engage with the complexities of our shared past. You can\r
	experience it yourself by visiting the portal at\r
\r
	`),P(e.$$.fragment,s),n=w(s,`\r
	.`)},m(s,o){p(s,t,o),A(e,s,o),p(s,n,o),i=!0},p(s,o){const f={};o&1&&(f.$$scope={dirty:o,ctx:s}),e.$set(f)},i(s){i||(y(e.$$.fragment,s),i=!0)},o(s){v(e.$$.fragment,s),i=!1},d(s){s&&(c(t),c(n)),I(e,s)}}}function ct(r){let t;return{c(){t=g("Read the original press release here.")},l(e){t=w(e,"Read the original press release here.")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function ft(r){let t,e;return t=new Q({props:{target:"_blank",href:"https://www.uni-kiel.de/en/details/news/128-roots-compass",$$slots:{default:[ct]},$$scope:{ctx:r}}}),{c(){k(t.$$.fragment)},l(n){P(t.$$.fragment,n)},m(n,i){A(t,n,i),e=!0},p(n,i){const s={};i&1&&(s.$$scope={dirty:i,ctx:n}),t.$set(s)},i(n){e||(y(t.$$.fragment,n),e=!0)},o(n){v(t.$$.fragment,n),e=!1},d(n){I(t,n)}}}function ut(r){let t,e,n,i,s,o,f,$,h,b;return t=new C({props:{$$slots:{default:[rt]},$$scope:{ctx:r}}}),n=new C({props:{$$slots:{default:[at]},$$scope:{ctx:r}}}),s=new C({props:{$$slots:{default:[it]},$$scope:{ctx:r}}}),f=new C({props:{$$slots:{default:[lt]},$$scope:{ctx:r}}}),h=new C({props:{$$slots:{default:[ft]},$$scope:{ctx:r}}}),{c(){k(t.$$.fragment),e=S(),k(n.$$.fragment),i=S(),k(s.$$.fragment),o=S(),k(f.$$.fragment),$=S(),k(h.$$.fragment)},l(u){P(t.$$.fragment,u),e=x(u),P(n.$$.fragment,u),i=x(u),P(s.$$.fragment,u),o=x(u),P(f.$$.fragment,u),$=x(u),P(h.$$.fragment,u)},m(u,_){A(t,u,_),p(u,e,_),A(n,u,_),p(u,i,_),A(s,u,_),p(u,o,_),A(f,u,_),p(u,$,_),A(h,u,_),b=!0},p(u,[_]){const T={};_&1&&(T.$$scope={dirty:_,ctx:u}),t.$set(T);const O={};_&1&&(O.$$scope={dirty:_,ctx:u}),n.$set(O);const L={};_&1&&(L.$$scope={dirty:_,ctx:u}),s.$set(L);const B={};_&1&&(B.$$scope={dirty:_,ctx:u}),f.$set(B);const z={};_&1&&(z.$$scope={dirty:_,ctx:u}),h.$set(z)},i(u){b||(y(t.$$.fragment,u),y(n.$$.fragment,u),y(s.$$.fragment,u),y(f.$$.fragment,u),y(h.$$.fragment,u),b=!0)},o(u){v(t.$$.fragment,u),v(n.$$.fragment,u),v(s.$$.fragment,u),v(f.$$.fragment,u),v(h.$$.fragment,u),b=!1},d(u){u&&(c(e),c(i),c(o),c($)),I(t,u),I(n,u),I(s,u),I(f,u),I(h,u)}}}let pt=class extends X{constructor(t){super(),Z(this,t,null,ut,Y,{})}};const ht=new $e("Bringing the Past to Life","The Launch of the ROOTS Compass Portal",new Date(Date.parse("22 August 2024")),`${ce}/media/roots-compass/map_view.webp`,`I am excited to share the launch of the ROOTS Compass,
    an innovative online platform that offers interactive access
    to the wide range of multidisciplinary research carried out
    by the Cluster of Excellence ROOTS.`,pt);function mt(r){let t;return{c(){t=g("Prof. Dr. Matthias Müller-Hannemann")},l(e){t=w(e,"Prof. Dr. Matthias Müller-Hannemann")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function $t(r){let t;return{c(){t=g("Dr. Steffen Schüler")},l(e){t=w(e,"Dr. Steffen Schüler")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function dt(r){let t;return{c(){t=g("Antonia Schmidt")},l(e){t=w(e,"Antonia Schmidt")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function gt(r){let t,e,n,i,s,o,f,$;return e=new Q({props:{href:"https://www.informatik.uni-halle.de/arbeitsgruppen/datenstrukturen/mitarbeiter/muellerh/",target:"_blank",$$slots:{default:[mt]},$$scope:{ctx:r}}}),i=new Q({props:{href:"https://www.informatik.uni-halle.de/arbeitsgruppen/datenstrukturen/mitarbeiter/steffen_schueler/",target:"_blank",$$slots:{default:[$t]},$$scope:{ctx:r}}}),o=new Q({props:{href:"https://www.informatik.uni-halle.de/arbeitsgruppen/datenstrukturen/mitarbeiter/schmidt/",target:"_blank",$$slots:{default:[dt]},$$scope:{ctx:r}}}),{c(){t=g(`Even more exciting, the paper was honored with the Best Paper Award at the conference! This\r
	achievement marks a major milestone in my professional journey, as the work is primarily derived\r
	from the results of my bachelor’s thesis. The research was guided by my excellent supervisors,\r
	`),k(e.$$.fragment),n=g(`\r
	and\r
	`),k(i.$$.fragment),s=g(`\r
	, from the Data Structures and Efficient Algorithms group at Martin Luther University Halle-Wittenberg.\r
	I was also fortunate to collaborate with brilliant researchers\r
	`),k(o.$$.fragment),f=g(`\r
	and Robert Mank, whose contributions were instrumental to our success.`)},l(h){t=w(h,`Even more exciting, the paper was honored with the Best Paper Award at the conference! This\r
	achievement marks a major milestone in my professional journey, as the work is primarily derived\r
	from the results of my bachelor’s thesis. The research was guided by my excellent supervisors,\r
	`),P(e.$$.fragment,h),n=w(h,`\r
	and\r
	`),P(i.$$.fragment,h),s=w(h,`\r
	, from the Data Structures and Efficient Algorithms group at Martin Luther University Halle-Wittenberg.\r
	I was also fortunate to collaborate with brilliant researchers\r
	`),P(o.$$.fragment,h),f=w(h,`\r
	and Robert Mank, whose contributions were instrumental to our success.`)},m(h,b){p(h,t,b),A(e,h,b),p(h,n,b),A(i,h,b),p(h,s,b),A(o,h,b),p(h,f,b),$=!0},p(h,b){const u={};b&1&&(u.$$scope={dirty:b,ctx:h}),e.$set(u);const _={};b&1&&(_.$$scope={dirty:b,ctx:h}),i.$set(_);const T={};b&1&&(T.$$scope={dirty:b,ctx:h}),o.$set(T)},i(h){$||(y(e.$$.fragment,h),y(i.$$.fragment,h),y(o.$$.fragment,h),$=!0)},o(h){v(e.$$.fragment,h),v(i.$$.fragment,h),v(o.$$.fragment,h),$=!1},d(h){h&&(c(t),c(n),c(s),c(f)),I(e,h),I(i,h),I(o,h)}}}function wt(r){let t;return{c(){t=g("The Problem: Spatial Transcriptomics and Barcode Layout Optimization")},l(e){t=w(e,"The Problem: Spatial Transcriptomics and Barcode Layout Optimization")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function bt(r){let t;return{c(){t=g(`Our paper addresses a unique and challenging optimization problem in the context of spatial\r
	transcriptomics, an exciting field that enables high-resolution spatial mapping of gene\r
	expression. At the heart of this problem is the selection and layout of barcodes—short DNA\r
	sequences that act as unique identifiers within a two-dimensional microarray. The arrays in\r
	question are enormous, with typical dimensions of 768×1024, which means over 786,000 barcodes need\r
	to be placed optimally.`)},l(e){t=w(e,`Our paper addresses a unique and challenging optimization problem in the context of spatial\r
	transcriptomics, an exciting field that enables high-resolution spatial mapping of gene\r
	expression. At the heart of this problem is the selection and layout of barcodes—short DNA\r
	sequences that act as unique identifiers within a two-dimensional microarray. The arrays in\r
	question are enormous, with typical dimensions of 768×1024, which means over 786,000 barcodes need\r
	to be placed optimally.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function _t(r){let t;return{c(){t=g(`The goal is to assign barcodes to positions on the microarray in a way that minimizes a\r
	position-dependent cost function. In essence, we need to solve a very specialized version of the\r
	quadratic assignment problem (QAP), which is known for its computational complexity. Not only is\r
	the general QAP difficult, but we also demonstrated that the barcode layout problem is\r
	MaxSNP-hard, meaning it’s also hard to approximate.`)},l(e){t=w(e,`The goal is to assign barcodes to positions on the microarray in a way that minimizes a\r
	position-dependent cost function. In essence, we need to solve a very specialized version of the\r
	quadratic assignment problem (QAP), which is known for its computational complexity. Not only is\r
	the general QAP difficult, but we also demonstrated that the barcode layout problem is\r
	MaxSNP-hard, meaning it’s also hard to approximate.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function yt(r){let t;return{c(){t=g("Approach and Results: Heuristics and CUDA for Faster Runtime")},l(e){t=w(e,"Approach and Results: Heuristics and CUDA for Faster Runtime")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function vt(r){let t,e,n="<li>A sorting-based algorithm</li> <li>A greedy algorithm</li> <li>2-OPT-based local search</li> <li>A genetic algorithm</li>",i;return{c(){t=g(`Given the complexity of the problem, exact methods like Integer Linear Programming (ILP) are\r
	infeasible for large instances—think solving for over 786,000 variables! While ILP can\r
	theoretically yield optimal results, it’s only applicable to very small-scale cases. To tackle\r
	realistic, large-scale instances, we developed a set of heuristics, including:\r
	`),e=j("ul"),e.innerHTML=n,i=g(`\r
	Among these, we found that a combination of the greedy algorithm and 2-OPT local search yielded the\r
	best results in terms of both solution quality and runtime. On large-scale arrays, our methods achieved\r
	an impressive 37% reduction in cost compared to random layouts and computed solutions in under 3 minutes.\r
	This was made possible thanks to CUDA-based optimizations, which allowed us to significantly accelerate\r
	runtime by utilizing GPU parallel processing. By offloading computation to GPUs, we were able to efficiently\r
	handle the large-scale nature of the problem while maintaining high-quality results.`),this.h()},l(s){t=w(s,`Given the complexity of the problem, exact methods like Integer Linear Programming (ILP) are\r
	infeasible for large instances—think solving for over 786,000 variables! While ILP can\r
	theoretically yield optimal results, it’s only applicable to very small-scale cases. To tackle\r
	realistic, large-scale instances, we developed a set of heuristics, including:\r
	`),e=U(s,"UL",{class:!0,"data-svelte-h":!0}),Se(e)!=="svelte-obtrjr"&&(e.innerHTML=n),i=w(s,`\r
	Among these, we found that a combination of the greedy algorithm and 2-OPT local search yielded the\r
	best results in terms of both solution quality and runtime. On large-scale arrays, our methods achieved\r
	an impressive 37% reduction in cost compared to random layouts and computed solutions in under 3 minutes.\r
	This was made possible thanks to CUDA-based optimizations, which allowed us to significantly accelerate\r
	runtime by utilizing GPU parallel processing. By offloading computation to GPUs, we were able to efficiently\r
	handle the large-scale nature of the problem while maintaining high-quality results.`),this.h()},h(){q(e,"class","list-disc pl-8")},m(s,o){p(s,t,o),p(s,e,o),p(s,i,o)},p:le,d(s){s&&(c(t),c(e),c(i))}}}function Tt(r){let t;return{c(){t=g("Exploiting Barcode Surplus and Biochemical Constraints")},l(e){t=w(e,"Exploiting Barcode Surplus and Biochemical Constraints")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function kt(r){let t;return{c(){t=g(`One particularly interesting aspect of our research was the surplus of barcodes. Since the pool of\r
	possible barcodes is much larger than the number actually needed, we explored the impact of\r
	selecting barcodes from this larger set. By incorporating surplus barcodes, we were able to\r
	significantly improve the layout quality, though this came with a slight increase in runtime.`)},l(e){t=w(e,`One particularly interesting aspect of our research was the surplus of barcodes. Since the pool of\r
	possible barcodes is much larger than the number actually needed, we explored the impact of\r
	selecting barcodes from this larger set. By incorporating surplus barcodes, we were able to\r
	significantly improve the layout quality, though this came with a slight increase in runtime.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Pt(r){let t;return{c(){t=g(`Another noteworthy discovery was that biochemical constraints, which limit the barcode design\r
	space, were actually beneficial for the overall layout cost. These constraints, rather than\r
	hindering the process, helped reduce complexity in the layout optimization, leading to better\r
	solutions.`)},l(e){t=w(e,`Another noteworthy discovery was that biochemical constraints, which limit the barcode design\r
	space, were actually beneficial for the overall layout cost. These constraints, rather than\r
	hindering the process, helped reduce complexity in the layout optimization, leading to better\r
	solutions.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function At(r){let t;return{c(){t=g("Conclusion and Looking Ahead")},l(e){t=w(e,"Conclusion and Looking Ahead")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function It(r){let t;return{c(){t=g(`The journey from my bachelor’s thesis to an award-winning paper has been both challenging and\r
	rewarding. I’m deeply grateful for the guidance and support from my supervisors and collaborators,\r
	as well as the opportunity to present this work at SEA 2024 in Vienna. The results of our research\r
	show promising advances in solving large-scale combinatorial optimization problems, and I’m\r
	excited about future applications in spatial transcriptomics and beyond.`)},l(e){t=w(e,`The journey from my bachelor’s thesis to an award-winning paper has been both challenging and\r
	rewarding. I’m deeply grateful for the guidance and support from my supervisors and collaborators,\r
	as well as the opportunity to present this work at SEA 2024 in Vienna. The results of our research\r
	show promising advances in solving large-scale combinatorial optimization problems, and I’m\r
	excited about future applications in spatial transcriptomics and beyond.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function St(r){let t;return{c(){t=g(`If you’d like to dive deeper into the technical details, you can find the paper in the\r
				Leibniz International Proceedings in Informatics (LIPIcs) series`)},l(e){t=w(e,`If you’d like to dive deeper into the technical details, you can find the paper in the\r
				Leibniz International Proceedings in Informatics (LIPIcs) series`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function xt(r){let t;return{c(){t=g(`The CUDA optimized C++ implementations of the algorithms described in the paper can be found\r
				here.`)},l(e){t=w(e,`The CUDA optimized C++ implementations of the algorithms described in the paper can be found\r
				here.`)},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Ot(r){let t;return{c(){t=g("Find out more about the SEA24 conference!")},l(e){t=w(e,"Find out more about the SEA24 conference!")},m(e,n){p(e,t,n)},d(e){e&&c(t)}}}function Lt(r){let t,e,n,i,s,o,f,$,h,b,u,_;return n=new Q({props:{href:"https://doi.org/10.4230/LIPIcs.SEA.2024.17",target:"_blank",$$slots:{default:[St]},$$scope:{ctx:r}}}),f=new Q({props:{href:"https://github.com/uni-halle/BarLay",target:"_blank",$$slots:{default:[xt]},$$scope:{ctx:r}}}),u=new Q({props:{href:"https://sea2024.univie.ac.at/",target:"_blank",$$slots:{default:[Ot]},$$scope:{ctx:r}}}),{c(){t=j("ul"),e=j("li"),k(n.$$.fragment),i=g("."),s=S(),o=j("li"),k(f.$$.fragment),$=g("."),h=S(),b=j("li"),k(u.$$.fragment),this.h()},l(T){t=U(T,"UL",{class:!0});var O=N(t);e=U(O,"LI",{});var L=N(e);P(n.$$.fragment,L),i=w(L,"."),L.forEach(c),s=x(O),o=U(O,"LI",{});var B=N(o);P(f.$$.fragment,B),$=w(B,"."),B.forEach(c),h=x(O),b=U(O,"LI",{});var z=N(b);P(u.$$.fragment,z),z.forEach(c),O.forEach(c),this.h()},h(){q(t,"class","list-disc pl-8")},m(T,O){p(T,t,O),D(t,e),A(n,e,null),D(e,i),D(t,s),D(t,o),A(f,o,null),D(o,$),D(t,h),D(t,b),A(u,b,null),_=!0},p(T,O){const L={};O&1&&(L.$$scope={dirty:O,ctx:T}),n.$set(L);const B={};O&1&&(B.$$scope={dirty:O,ctx:T}),f.$set(B);const z={};O&1&&(z.$$scope={dirty:O,ctx:T}),u.$set(z)},i(T){_||(y(n.$$.fragment,T),y(f.$$.fragment,T),y(u.$$.fragment,T),_=!0)},o(T){v(n.$$.fragment,T),v(f.$$.fragment,T),v(u.$$.fragment,T),_=!1},d(T){T&&c(t),I(n),I(f),I(u)}}}function zt(r){let t,e,n,i,s,o,f,$,h,b,u,_,T,O,L,B,z,M,E,G,R,W,H,J;return t=new C({props:{$$slots:{default:[gt]},$$scope:{ctx:r}}}),n=new K({props:{$$slots:{default:[wt]},$$scope:{ctx:r}}}),s=new C({props:{$$slots:{default:[bt]},$$scope:{ctx:r}}}),f=new C({props:{$$slots:{default:[_t]},$$scope:{ctx:r}}}),h=new K({props:{$$slots:{default:[yt]},$$scope:{ctx:r}}}),u=new C({props:{$$slots:{default:[vt]},$$scope:{ctx:r}}}),T=new K({props:{$$slots:{default:[Tt]},$$scope:{ctx:r}}}),L=new C({props:{$$slots:{default:[kt]},$$scope:{ctx:r}}}),z=new C({props:{$$slots:{default:[Pt]},$$scope:{ctx:r}}}),E=new K({props:{$$slots:{default:[At]},$$scope:{ctx:r}}}),R=new C({props:{$$slots:{default:[It]},$$scope:{ctx:r}}}),H=new C({props:{$$slots:{default:[Lt]},$$scope:{ctx:r}}}),{c(){k(t.$$.fragment),e=S(),k(n.$$.fragment),i=S(),k(s.$$.fragment),o=S(),k(f.$$.fragment),$=S(),k(h.$$.fragment),b=S(),k(u.$$.fragment),_=S(),k(T.$$.fragment),O=S(),k(L.$$.fragment),B=S(),k(z.$$.fragment),M=S(),k(E.$$.fragment),G=S(),k(R.$$.fragment),W=S(),k(H.$$.fragment)},l(l){P(t.$$.fragment,l),e=x(l),P(n.$$.fragment,l),i=x(l),P(s.$$.fragment,l),o=x(l),P(f.$$.fragment,l),$=x(l),P(h.$$.fragment,l),b=x(l),P(u.$$.fragment,l),_=x(l),P(T.$$.fragment,l),O=x(l),P(L.$$.fragment,l),B=x(l),P(z.$$.fragment,l),M=x(l),P(E.$$.fragment,l),G=x(l),P(R.$$.fragment,l),W=x(l),P(H.$$.fragment,l)},m(l,d){A(t,l,d),p(l,e,d),A(n,l,d),p(l,i,d),A(s,l,d),p(l,o,d),A(f,l,d),p(l,$,d),A(h,l,d),p(l,b,d),A(u,l,d),p(l,_,d),A(T,l,d),p(l,O,d),A(L,l,d),p(l,B,d),A(z,l,d),p(l,M,d),A(E,l,d),p(l,G,d),A(R,l,d),p(l,W,d),A(H,l,d),J=!0},p(l,[d]){const F={};d&1&&(F.$$scope={dirty:d,ctx:l}),t.$set(F);const ee={};d&1&&(ee.$$scope={dirty:d,ctx:l}),n.$set(ee);const V={};d&1&&(V.$$scope={dirty:d,ctx:l}),s.$set(V);const te={};d&1&&(te.$$scope={dirty:d,ctx:l}),f.$set(te);const a={};d&1&&(a.$$scope={dirty:d,ctx:l}),h.$set(a);const m={};d&1&&(m.$$scope={dirty:d,ctx:l}),u.$set(m);const ne={};d&1&&(ne.$$scope={dirty:d,ctx:l}),T.$set(ne);const se={};d&1&&(se.$$scope={dirty:d,ctx:l}),L.$set(se);const re={};d&1&&(re.$$scope={dirty:d,ctx:l}),z.$set(re);const ae={};d&1&&(ae.$$scope={dirty:d,ctx:l}),E.$set(ae);const ie={};d&1&&(ie.$$scope={dirty:d,ctx:l}),R.$set(ie);const oe={};d&1&&(oe.$$scope={dirty:d,ctx:l}),H.$set(oe)},i(l){J||(y(t.$$.fragment,l),y(n.$$.fragment,l),y(s.$$.fragment,l),y(f.$$.fragment,l),y(h.$$.fragment,l),y(u.$$.fragment,l),y(T.$$.fragment,l),y(L.$$.fragment,l),y(z.$$.fragment,l),y(E.$$.fragment,l),y(R.$$.fragment,l),y(H.$$.fragment,l),J=!0)},o(l){v(t.$$.fragment,l),v(n.$$.fragment,l),v(s.$$.fragment,l),v(f.$$.fragment,l),v(h.$$.fragment,l),v(u.$$.fragment,l),v(T.$$.fragment,l),v(L.$$.fragment,l),v(z.$$.fragment,l),v(E.$$.fragment,l),v(R.$$.fragment,l),v(H.$$.fragment,l),J=!1},d(l){l&&(c(e),c(i),c(o),c($),c(b),c(_),c(O),c(B),c(M),c(G),c(W)),I(t,l),I(n,l),I(s,l),I(f,l),I(h,l),I(u,l),I(T,l),I(L,l),I(z,l),I(E,l),I(R,l),I(H,l)}}}class Et extends X{constructor(t){super(),Z(this,t,null,zt,Y,{})}}const Dt=new $e("Publishing my First Paper","Barcode Selection and Layout Optimization in Spatial Transcriptomics",new Date(Date.parse("24 Juli 2024")),`${ce}/media/sea24/photolitography.png`,`I’m thrilled to announce that
	my recent paper, “Barcode Selection and Layout Optimization in Spatial Transcriptomics,” has been
	published in the proceedings of the 22nd International Symposium on Experimental Algorithms (SEA
	2024), which took place in beautiful Vienna, Austria.`,Et),Bt=[ht,Dt,nt].sort((r,t)=>t.date.getTime()-r.date.getTime()),qt=new Map;for(const r of Bt)qt.set(r.identifier,r);function Ct(r){let t,e,n,i=r[0].date.toLocaleDateString()+"",s,o,f,$=r[0].title+"",h,b,u,_,T,O,L=r[0].intro+"",B,z;return{c(){t=j("a"),e=j("div"),n=j("div"),s=g(i),o=S(),f=j("div"),h=g($),b=S(),u=j("img"),T=S(),O=j("p"),B=g(L),this.h()},l(M){t=U(M,"A",{class:!0,id:!0,href:!0});var E=N(t);e=U(E,"DIV",{class:!0});var G=N(e);n=U(G,"DIV",{class:!0});var R=N(n);s=w(R,i),R.forEach(c),o=x(G),f=U(G,"DIV",{class:!0});var W=N(f);h=w(W,$),W.forEach(c),b=x(G),u=U(G,"IMG",{src:!0}),T=x(G),O=U(G,"P",{class:!0});var H=N(O);B=w(H,L),H.forEach(c),G.forEach(c),E.forEach(c),this.h()},h(){q(n,"class","text-xs text-purple-900"),q(f,"class","text-xl mb-4 text-purple-900"),Ie(u.src,_=r[0].heroImage)||q(u,"src",_),q(O,"class","text-purple-900 text-base mt-4"),q(e,"class","px-6 py-4"),q(t,"class","max-w-sm overflow-hidden shadow-lg bg-amber-400 col-span-1 svelte-wdz3tv"),q(t,"id","outer"),q(t,"href",z=`${Ae}/blog/${encodeURIComponent(r[0].identifier)}`)},m(M,E){p(M,t,E),D(t,e),D(e,n),D(n,s),D(e,o),D(e,f),D(f,h),D(e,b),D(e,u),D(e,T),D(e,O),D(O,B)},p(M,[E]){E&1&&i!==(i=M[0].date.toLocaleDateString()+"")&&me(s,i),E&1&&$!==($=M[0].title+"")&&me(h,$),E&1&&!Ie(u.src,_=M[0].heroImage)&&q(u,"src",_),E&1&&L!==(L=M[0].intro+"")&&me(B,L),E&1&&z!==(z=`${Ae}/blog/${encodeURIComponent(M[0].identifier)}`)&&q(t,"href",z)},i:le,o:le,d(M){M&&c(t)}}}function Mt(r,t,e){let{post:n}=t;return r.$$set=i=>{"post"in i&&e(0,n=i.post)},[n]}class Qt extends X{constructor(t){super(),Z(this,t,Mt,Ct,Y,{post:0})}}function jt(r){let t,e,n,i;const s=r[1].default,o=fe(s,r,r[0],null);return{c(){t=j("h2"),o&&o.c(),e=S(),n=j("div"),this.h()},l(f){t=U(f,"H2",{class:!0});var $=N(t);o&&o.l($),e=x($),n=U($,"DIV",{class:!0}),N(n).forEach(c),$.forEach(c),this.h()},h(){q(n,"class","h-px bg-amber-400 w-1/4 mt-1"),q(t,"class","font-sans text-2xl text-left py-2 text-purple-900 mb-4")},m(f,$){p(f,t,$),o&&o.m(t,null),D(t,e),D(t,n),i=!0},p(f,[$]){o&&o.p&&(!i||$&1)&&ue(o,s,f,f[0],i?he(s,f[0],$,null):pe(f[0]),null)},i(f){i||(y(o,f),i=!0)},o(f){v(o,f),i=!1},d(f){f&&c(t),o&&o.d(f)}}}function Ut(r,t,e){let{$$slots:n={},$$scope:i}=t;return r.$$set=s=>{"$$scope"in s&&e(0,i=s.$$scope)},[i,n]}class Jt extends X{constructor(t){super(),Z(this,t,Ut,jt,Y,{})}}export{Q as A,Qt as B,Jt as H,qt as M,C as P,Bt as a,Wt as e};
